### DISCUSSION TITLE
PyVistaQt Drag and Drop

### DISCUSSION THREAD

**Discussion by ndahiya3** [pyvista/pyvista#3280] [Q&A]
I am using pyvistaqt QtInteractor as the central widget of Qt application. I have menus/widgets to load surfaces/points etc. and manipulate them. QtInteractor plotter seems to have Drag and Drop auto enabled. But it bypasses the Qt App, so if I drag and drop something it get's displayed in the plotter but my Qt Main Window class is unaware of the added surface and I can't call any of my processing routines on the surface.

Is there any way to get around this? Maybe disable drag/drop from QtInteractor and implement that in the QMainWindow class where I can connect the Drag/Drop to my functions?

Any help on the topic is appreciated.

**Comment by ndahiya3** [comment]
If I add the following to my QMainWindow class:
```
def dragEnterEvent(self, e):
    print(e)
		
    if e.mimeData().hasUrls():
       e.accept()
    else:
       e.ignore()
         
  def dropEvent(self, e):
    files = [u.toLocalFile() for u in e.mimeData().urls()]
    for f in files:
      print(f)
```
then the dragEnterEvent is called but not the dropEvent. I think it is because the QtInteractor handles the drop event and it is never passed to QMainWindow. However, if I do something like this:
```
self.plotter = QtInteractor(self.frame)
self.plotter.interactor.setAcceptDrops(False) <----
```
then the QMainWindow dropEvent is called and the QtInteractor does nothing. So I think the problem is solved, unless I run into something else!

**Reply by banesullivan** [reply]
cc @pyvista/pyvistaqt-admin

**Reply by larsoner** [reply]
Disabling drops for the `interactor` so that they go through to the MainWindow seems reasonable to me
